tensorboard/webapp/feature_flag/views/feature_flag_modal_trigger_component.ts

```diff
@@ -0,0 +1,54 @@
+/* Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component} from '@angular/core';
+import {MatDialog, MatDialogRef} from '@angular/material/dialog';
+import {Store} from '@ngrx/store';
+import {State} from '../../app_state';
+import {featureFlagOverrideChanged} from '../actions/feature_flag_actions';
+import {
+  getFeatureFlags,
+  getShowFlagsEnabled,
+} from '../store/feature_flag_selectors';
+import {FeatureFlagPageContainer} from '../views/feature_flag_page_container';
+
+@Component({
+  selector: 'feature-flag-modal-trigger-component',
+  template: ``,
+  styles: [],
+})
+export class FeatureFlagModalTriggerComponent {
+  readonly showFeatureFlags$ = this.store.select(getShowFlagsEnabled);
+  readonly featureFlags$ = this.store.select(getFeatureFlags);
+  private featureFlagsDialog?: MatDialogRef<FeatureFlagPageContainer>;
+
+  constructor(private readonly store: Store<State>, private dialog: MatDialog) {
+    this.showFeatureFlags$.subscribe((showFeatureFlags: boolean) => {
+      if (showFeatureFlags) {
+        this.featureFlagsDialog = this.dialog.open(FeatureFlagPageContainer);
+        this.featureFlagsDialog.afterClosed().subscribe(() => {
+          this.store.dispatch(
+            featureFlagOverrideChanged({
```

Comment:
Interesting. Can you add a comment explaining why you do this? (I guess you are just trying to keep the dialog from reopening?)

Reply:
I'll go ahead and add a comment. This is to stop the dialog from opening again after the page is refreshed as I imagine people do not want this to always appear.